### DISCUSSION TITLE
Configuring with pgcli.conf on Windows 10

### DISCUSSION THREAD

**Discussion by shortthirdman** [dbcli/pgcli#1329] [Q&A] [answered]
I have configured the PGCLI with the `pgcli.conf` and have put it on the user's home directory. But still no changes have been observed like - still I have to pass on the connection details and even no changes on the prompt string.

What is that I am missing in configuration?

Guys, please help me out.

**Comment by j-bennet** [comment, marked as answer]
In Windows, pgcli [stores the config in](https://github.com/dbcli/pgcli/blob/366c01bbb97fee96a8169629854af7c83d626c99/pgcli/config.py#L15):

`%USERPROFILE%\\AppData\\Local\\dbcli\\pgcli\\`

Where `USERPROFILE` expands to `C:\\Users\your_username`.

If you already ran `pgcli`, it should have written the default config there.

**Reply by shortthirdman** [reply]
I had got to the location after posting this question.

**Comment by shortthirdman** [comment]
But I am facing issue in configuring DSNs in the config file for PostgreSQL.
like 

`netstar01 = postgresql://postgres:admin#pgsql@localhost:5432/nsprdb01`

It doesn't parse the password well which in this case is `admin#pgsql` and instead takes in database user ( `postgres` ) as the host name.

**Reply by dvarrazzo** [reply]
You have to [urlquote](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote) the password if you are composing a connection string as url.

**Reply by shortthirdman** [reply]
Thanks for the suggestion.
